Fix null error in InstallationContextConfig

ValgulNecron opened the pull request:
Added #[serde(default)] to oauth2_install_params in InstallationContextConfig and added a Default implementation to InstallParams #2883 and https://discord.com/developers/docs/resources/application#application-object-application-structure.

ValgulNecron committed:
Fix null error in InstallationContextConfig

Added #[serde(default)] to oauth2_install_params in InstallationContextConfig and added a Default implementation to InstallParams
#2883 and https://discord.com/developers/docs/resources/application#application-object-application-structure.

mkrasnitski commented on src/model/application/mod.rs:
```diff
@@ -124,7 +124,8 @@ enum_number! {
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct InstallationContextConfig {
-    pub oauth2_install_params: InstallParams,
+    #[serde(default)]
+    pub oauth2_install_params: InstallParams, // TODO: Replace with an option instead of #[serde(default)]
```
```suggestion
    // TODO: Replace with an option instead of #[serde(default)]
    #[serde(default)]
    pub oauth2_install_params: InstallParams,
```
Formatting nit

mkrasnitski commented:
Since `CurrentApplicationInfo::integration_types_config` is gated behind `unstable_discord_api`, it's actually ok to break the serenity API and change its `oauth2_install_params` field to `Option<InstallParams>`.

ValgulNecron committed:
Changed to an option

reverted back the change I made in the last commit and wrapped InstallParams in an Option

ValgulNecron commented:
Hi thanks for the info i changed it to an option and removed the change I made before
